src/main/java/org/reaktivity/reaktor/Reaktor.java

```diff
@@ -110,6 +111,11 @@ public void detach(
         }
     }
 
+    public void visit(Consumer<Controller> action)
+    {
+        this.controllersByKind.values().forEach(action);
+    }
+
```

Comment:
It would also be sufficient to add the following method to `Reaktor`.

``` java
public Set<Class<? extends Controller>> controllerKinds()
{
  return controllersByKind.keySet();
}
```
Then use `reaktor.controller(kind)` to get the specific `Controller` instances.

For completeness, also add:
``` java
public Set<String> nukleusNames()
{
  return nukleiByName.keySet();
}
```